connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/LiteralValueProtoConverter.scala

```diff
@@ -130,4 +135,117 @@ object LiteralValueProtoConverter {
       case o => throw new Exception(s"Unsupported value type: $o")
     }
   }
+
+  private def toArrayData(array: proto.Expression.Literal.Array): Any = {
```

Comment:
Can you use a scala ArrayBuilder here? You will need a classTag for that tough.

Reply:
Let me try